larreco/RecoAlg/TrackMomentumCalculator.cxx

```diff
@@ -265,6 +437,12 @@ namespace trkf {
 
     if (recoX.size() < 2) return -1.0;
 
+    // If set to true, scattered angles in segments with only 2 points will not be considered
+    check_valid_scattered = check_valid_scattered_;
+
+    // Correction due to oversmoothing, applied for space angle only
+    angle_correction = angle_correction_;
+
```

Comment:
Instead of setting class variables in a member function, please pass `check_valid_scattered_` and `angle_correction_` to the functions that need it (e.g. `getDeltaThetaij_`).